client/components/domains/register-domain-step/index.jsx

```diff
@@ -292,29 +309,44 @@ const RegisterDomainStep = React.createClass( {
 						}
 
 						this.props.onDomainsAvailabilityChange( true );
-
 						callback( null, result );
 					} );
 				},
 				callback => {
 					const query = {
-						query: domain,
-						quantity: SUGGESTION_QUANTITY,
-						include_wordpressdotcom: this.props.includeWordPressDotCom,
-						vendor: abtest( 'domainSuggestionVendor' )
-					};
-
+							query: domain,
+							quantity: SUGGESTION_QUANTITY,
+							include_wordpressdotcom: this.props.includeWordPressDotCom,
+							vendor: abtest( 'domainSuggestionVendor' )
+						},
+						timestamp = Date.now();
 					domains.suggestions( query ).then( domainSuggestions => {
 						this.props.onDomainsAvailabilityChange( true );
+						const timeDiff = Date.now() - timestamp;
+						this.recordEvent(
+							'searchResultsReceive',
+							domain,
+							domainSuggestions.map( suggestion => suggestion.domain_name ),
+							timeDiff,
+							domainSuggestions.length,
+							this.props.analyticsSection
+						);
 						callback( null, domainSuggestions );
 					} ).catch( error => {
+						const timeDiff = Date.now() - timestamp;
 						if ( error && error.statusCode === 503 ) {
-							return this.props.onDomainsAvailabilityChange( false );
+							this.props.onDomainsAvailabilityChange( false );
```

Comment:
Not entirely sure about this, but it seems we'd still need to call `callback` in the end.